distributed/shuffle/_shuffle.py

```diff
@@ -1,20 +1,23 @@
 from __future__ import annotations
 
 import logging
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, ClassVar, NewType
 
 from dask.base import tokenize
 from dask.highlevelgraph import HighLevelGraph
 from dask.layers import SimpleShuffleLayer
 
-from distributed.shuffle._shuffle_extension import ShuffleId, ShuffleWorkerExtension
-
 logger = logging.getLogger("distributed.shuffle")
 if TYPE_CHECKING:
     import pandas as pd
 
     from dask.dataframe import DataFrame
 
+    # circular dependency
+    from distributed.shuffle._worker_extension import ShuffleWorkerExtension
+
+ShuffleId = NewType("ShuffleId", str)
```

Comment:
I've moved this into `_shuffle` now since the `P2PShuffleLayer` is responsible for all ID- or naming-related things. Some helpers from the `ShuffleSchedulerExtension` moved here as well which allows us to rely on constants instead of hard-coded strings withing methods.

Moving `ShuffleId` into `__init__` would create some cyclic typing dependencies that I'm not too fond of.